### PR TITLE
Use generate_bundle.sh to create the CSV for stf-run-ci role

### PR DESCRIPTION
**Pull request by leifmadsen**
Depends on https://github.com/infrawatch/smart-gateway-operator/pull/83

### CODE REVIEW

**Comment by leifmadsen** [comment]
This is _almost_ working but the CSV generated for some reason provides a double tag on the end, so gotta figure that out.

**Comment by leifmadsen** [comment]
Depends on https://github.com/infrawatch/smart-gateway-operator/pull/83

**Comment by leifmadsen** [comment]
It works!

```
TASK [stf-run-ci : debug] ********************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "validate_deployment.stdout_lines": [
        "",
        "* [info] Waiting for QDR deployment to complete",
        "",
        "Waiting for deployment \"default-interconnect\" rollout to finish: 0 of 1 updated replicas are available...",
        "deployment \"default-interconnect\" successfully rolled out",
        "",
        "* [info] Waiting for prometheus deployment to complete",
        "",
        "Waiting for 1 pods to be ready...",
        "statefulset rolling update complete 1 pods at revision prometheus-default-85f68646b...",
        "",
        "* [info] Waiting for elasticsearch deployment to complete ",
        "",
        "",
        "* [info] Waiting for alertmanager deployment to complete",
        "",
        "statefulset rolling update complete 1 pods at revision alertmanager-default-fcb74598f...",
        "",
        "* [info] Waiting for smart-gateway deployment to complete",
        "",
        "Waiting for deployment \"default-cloud1-coll-meter-smartgateway\" rollout to finish: 0 of 1 updated replicas are available...",
        "deployment \"default-cloud1-coll-meter-smartgateway\" successfully rolled out",
        "deployment \"default-cloud1-coll-event-smartgateway\" successfully rolled out",
        "Waiting for deployment \"default-cloud1-ceil-event-smartgateway\" rollout to finish: 0 of 1 updated replicas are available...",
        "deployment \"default-cloud1-ceil-event-smartgateway\" successfully rolled out",
        "deployment \"default-cloud1-ceil-meter-smartgateway\" successfully rolled out",
        "",
        "* [info] Waiting for all pods to show Ready/Complete",
        "",
        "",
        "* [info] CI Build complete. You can now run tests."
    ]
}

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
localhost                  : ok=100  changed=43   unreachable=0    failed=0    skipped=2    rescued=0    ignored=0   
```

**Comment by leifmadsen** [comment]
Testing command since there is a dependency on a change in SGO:

```sh
ansible-playbook -e sgo_branch=$(git branch --show-current) run-ci.yaml
```

**Review by pleimer — APPROVED** [review]

File: build/stf-run-ci/tasks/setup_stf_local_build.yml [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+  shell:
+    chdir: working/smart-gateway-operator/build
+    cmd: |
+      WORKING_DIR="{{ playbook_dir }}/working/smart-gateway-operator-bundle" \
+      RELATED_IMAGE_CORE_SMARTGATEWAY={{ sg_core_image_path.split(':')[:-1] | join(':') | quote }} \
+      RELATED_IMAGE_BRIDGE_SMARTGATEWAY={{ sg_bridge_image_path.split(':')[:-1] | join(':') | quote }} \
+      RELATED_IMAGE_CORE_SMARTGATEWAY_TAG={{ sg_core_image_path.split(':')[-1] | quote }} \
+      RELATED_IMAGE_BRIDGE_SMARTGATEWAY_TAG={{ sg_bridge_image_path.split(':')[-1] | quote }} \
+      OPERATOR_IMAGE={{ sgo_image_path.split(':')[:-1] | join(':') | quote }} \
+      OPERATOR_TAG={{ sgo_image_path.split(':')[-1] | quote }} \
```
[the remark this answers is not in the document]
Comment (csibbitt) [reply build/stf-run-ci/tasks/setup_stf_local_build.yml:14]: This looks like a decent case for two simple filter plugins to help with readability and maintenance.  `{{ sg_core_image_path | parse_image }}` and `{{ sg_core_image_path | parse_tag }}`

https://github.com/infrawatch/service-telemetry-operator/tree/master/roles/servicetelemetry/filter_plugins

File: build/stf-run-ci/tasks/setup_stf_local_build.yml [hunk 2]
```diff
@@ -8,62 +27,26 @@
...
-    path: working/service-telemetry-operator.clusterserviceversion.yaml
-    regexp: 'placeholder'
-    replace: '{{ namespace }}'
+- name: Generate Service Telemetry Operator CSV
+  shell:
+    chdir: "{{ playbook_dir }}"
+    cmd: |
+      WORKING_DIR="{{ playbook_dir }}/working/service-telemetry-operator-bundle" \
+      RELATED_IMAGE_PROMETHEUS_WEBHOOK_SNMP={{ prometheus_webhook_snmp_image_path.split(':')[:-1] | join(':') | quote }} \
+      RELATED_IMAGE_PROMETHEUS_WEBHOOK_SNMP_TAG={{ prometheus_webhook_snmp_image_path.split(':')[-1] | quote }} \
```
Comment (csibbitt) [inline build/stf-run-ci/tasks/setup_stf_local_build.yml:46]: More use cases for the filters

**Review by csibbitt — APPROVED** [review]
Suggest filters to prevent cross-eyed'ness when reading those envvar templates

File: build/stf-run-ci/tasks/setup_stf_local_build.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (leifmadsen) [reply build/stf-run-ci/tasks/setup_stf_local_build.yml:14]: Hrmmm... I guess I'll need to look at learning some Python :)

[the remark this answers is not in the document]
Comment (csibbitt) [reply build/stf-run-ci/tasks/setup_stf_local_build.yml:14]: Roughly...

```
def parse_image(path):
  return '"' + path.split(':')[:-1].join(':') + '"'
```


There is a bit more boilerplate that you can see in the existing filter plugin.

[the remark this answers is not in the document]
Comment (leifmadsen) [reply build/stf-run-ci/tasks/setup_stf_local_build.yml:14]: Ah awesome thanks! That should give me enough to be dangerous.

[the remark this answers is not in the document]
Comment (leifmadsen) [reply build/stf-run-ci/tasks/setup_stf_local_build.yml]: Done!

File: build/stf-run-ci/tasks/setup_stf_local_build.yml (hunk 2 above)
Comment (leifmadsen) [reply build/stf-run-ci/tasks/setup_stf_local_build.yml]: Done.

**Comment by leifmadsen** [comment]
Re-tested this with the new filters and it passes in the same manner as previous. Going to merge this.

**Pull request merged**